src/profile/components/Settings/UserData/index.tsx

```diff
@@ -0,0 +1,99 @@
+import useAsync from 'common/hooks/useAsync';
+import React, { useEffect, useRef } from 'react';
+
+import { Button, Card, TextField } from '@dotkomonline/design-system';
+import { IUserContext, UserContext } from 'authentication/providers/UserProvider';
+import { anonymizeUser, fetchUserData } from 'profile/api/gdpr';
+
+import style from './userdata.less';
+
+interface IUserCredentials {
+  username: string;
+  password: string;
+}
+
+export const UserData = () => {
+  const userContext = React.useContext<IUserContext>(UserContext);
+  const user = userContext.user;
+
+  if (!user) {
+    return <></>;
+  }
+
+  const [downloadUserDataState, setDownloadUserDataState] = React.useState(false);
+  const [userCredentials, setUserCredentials] = React.useState<IUserCredentials>({ username: '', password: '' });
+  const [deleteUserState, setDeleteUserState] = React.useState(false);
+
+  const aRef = useRef(null);
+  const [fileRef, setFileRef] = React.useState<null | string>(null);
+  const [userData, , userDataFetchStatus] = useAsync<File | null>(async () => {
+    if (downloadUserDataState) {
+      return new File([await JSON.stringify(fetchUserData(user))], `${user.profile.preferred_username}.json`, {
+        type: 'application/json',
+      });
+    }
+    return null;
+  }, [downloadUserDataState, user]);
+
+  const [, ,] = useAsync<unknown>(async () => {
+    if (deleteUserState) {
+      return await anonymizeUser(user, user.profile.sub, userCredentials.username, userCredentials.password);
+    }
+    return null;
+  }, [deleteUserState]);
+
+  useEffect(() => {
+    if (userData != null && userDataFetchStatus === 'resolved') {
+      const ref = URL.createObjectURL(userData);
+      setFileRef(ref);
+      return () => {
+        URL.revokeObjectURL(ref);
+        setFileRef(null);
+      };
+    }
+    return;
+  }, [userData]);
+
+  useEffect(() => {
+    if (aRef.current != null) {
```

Comment:
`!==`?

You're already checking if it's `null`, why the ts-ignore?

Reply:
the linter would still report it as a warning even tho I wrapped it in a null check.